### PR TITLE
Feature/create task

### PR DESCRIPTION
**Pull request by M4R1KU**
This closes none.

Summary of changes:
- ​added TaskService on the server

I made sure to:

- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines

Please review @jmesserli / cc: @outcobra/developers.

### CODE REVIEW

File: backend/src/main/kotlin/outcobra/server/service/internal/DefaultTaskService.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+
+    override fun readAllOpenTasksBySubject(subjectId: Long): List<TaskDto> {
+        val filter = QTask.task.progress.eq(100).and(QTask.task.subject.id.eq(subjectId))
+        return repository.findAll(filter).map { mapper.toDto(it) }
+    }
+
+    override fun readAllOpenTasksBySemester(semesterId: Long): List<TaskDto> {
+        val filter = QTask.task.progress.eq(100).and(QTask.task.subject.semester.id.eq(semesterId))
```
Comment (M4R1KU) [inline backend/src/main/kotlin/outcobra/server/service/internal/DefaultTaskService.kt]: Is a task open when it's progress is at 100% :thinking:

File: backend/src/main/kotlin/outcobra/server/controller/TaskController.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package outcobra.server.controller
+
+import org.springframework.web.bind.annotation.*
+import outcobra.server.model.dto.TaskDto
+import outcobra.server.service.TaskService
+import javax.inject.Inject
+
+/**
+ * Created by Vincent on 15.11.2016.
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/controller/TaskController.kt]: This needs to be fixed

File: backend/src/main/kotlin/outcobra/server/model/dto/TaskDto.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package outcobra.server.model.dto
+
+import outcobra.server.model.Subject
+import outcobra.server.model.interfaces.OutcobraDto
+import outcobra.server.model.interfaces.ParentLink
+import java.time.LocalDate
+
+/**
+ * Created by Vincent on 15.11.2016.
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/model/dto/TaskDto.kt]: Same here

File: backend/src/main/kotlin/outcobra/server/model/mapper/TaskDtoMapper.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import org.springframework.stereotype.Component
+import outcobra.server.model.Task
+import outcobra.server.model.dto.TaskDto
+import outcobra.server.model.interfaces.Mapper
+import outcobra.server.model.repository.SubjectRepository
+import javax.inject.Inject
+
+/**
+ * Created by Vincent on 25.11.2016.
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/model/mapper/TaskDtoMapper.kt]: Same

File: backend/src/main/kotlin/outcobra/server/model/mapper/TaskDtoMapper.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ */
+@Component
+open class TaskDtoMapper @Inject constructor(val subjectRepository: SubjectRepository) : Mapper<Task, TaskDto> {
+
+    override fun toDto(from: Task): TaskDto {
+        return TaskDto(from.id, from.subject.id, from.name, from.description, from.todoDate, from.dueDate, from.effort, from.progress)
+    }
+
+    override fun fromDto(from: TaskDto): Task {
+        val subject = subjectRepository.findOne(from.id)
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/model/mapper/TaskDtoMapper.kt]: This is def. wrong. We can't get a Subject with a TaskDto id

File: backend/src/main/kotlin/outcobra/server/service/TaskService.kt [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * Reads all [Task]s of which the dueDate is not expired from the given subject
+     * @param subjectId the id of the subject you want to use as a filter
+     *@return All [Task]s which have the dueDate after now
+     */
+    fun readAllOpenTasksBySubject(subjectId: Long): List<TaskDto>
+
+    /**
+     *Reads all [Task]s that are associated with a specific [Semester] and have their dueDate not expired
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/service/TaskService.kt]: Format code

File: backend/src/main/kotlin/outcobra/server/service/internal/DefaultTaskService.kt (hunk 1 above)
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/service/internal/DefaultTaskService.kt]: Yeah that's wrong

**Pull request merged**